engine/src/test/java/com/datatorrent/stram/client/AppPackageTest.java

```diff
@@ -95,22 +97,51 @@ public void testAppPackage() throws Exception
     Assert.assertEquals(System.getProperty("apex.version", "3.4.0"), json.getString("dtEngineVersion"));
     Assert.assertEquals("lib/*.jar", json.getJSONArray("classPath").getString(0));
 
-    JSONObject application1 = json.getJSONArray("applications").getJSONObject(0);
-    JSONObject application2 = json.getJSONArray("applications").getJSONObject(1);
+    // Test if there are fixed number of applications
+    Assert.assertEquals("There are two applications", 2, json.getJSONArray("applications").length());
 
     Map<String, JSONObject> apps = new HashMap<>();
+    for (int i = 0; i < json.getJSONArray("applications").length(); i++) {
```

Comment:
introduce applications = json.getJSONArray("applications") and applicaiton = applicaiton.getJSONObject(i);